virtool/account/data.py

```diff
@@ -133,7 +137,7 @@ async def get_keys(self, user_id: str) -> List[APIKey]:
         """
         cursor = self._db.keys.find({"user.id": user_id}, API_KEY_PROJECTION)
 
-        return [APIKey(**d) async for d in cursor]
+        return [await fetch_complete_key(self._db, d["id"]) async for d in cursor]
```

Comment:
This isn't very efficient.

I think you can do this in a single request. If that is not possible, at the very least `gather` should be used to execute your `fetch_complete_key` calls concurrently.